hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -8,60 +10,140 @@ module Hydra.Chain.Direct.Tx where
+
+-- TODO(SN): parameterize
+network :: Network
+network = Testnet
 
--- * Hydra Head transactions
+-- * Post Hydra Head transactions
 
-constructTx :: TxIn -> PostChainTx tx -> TxBodyContent BuildTx AlonzoEra
+-- | Construct the Head protocol transactions as Alonzo 'Tx'. Note that
+-- 'ValidatedTx' this produces an unbalanced, unsigned transaction and this type
+-- was used (in contrast to 'TxBody') to be able to express included datums,
+-- onto which at least the 'initTx' relies on.
+constructTx :: TxIn StandardCrypto -> PostChainTx tx -> ValidatedTx (AlonzoEra StandardCrypto)
 constructTx txIn = \case
   InitTx p -> initTx p txIn
   _ -> error "not implemented"
 
--- | Smart constructors for 'TxBodyContent' which should be used with
--- 'makeTransactionBodyAutoBalance' to balance and account for fees accordingly.
-
 -- | Create the init transaction from some 'HeadParameters' and a single UTXO
 -- which will be used for minting NFTs.
-initTx :: HeadParameters -> TxIn -> TxBodyContent BuildTx AlonzoEra
-initTx _ txIn =
-  TxBodyContent
-    { txIns = [(txIn, BuildTxWith (KeyWitness KeyWitnessForSpending))]
-    , txInsCollateral = TxInsCollateralNone
-    , txOuts = []
-    , txFee = TxFeeExplicit TxFeesExplicitInAlonzoEra 0
-    , txValidityRange = (TxValidityNoLowerBound, TxValidityNoUpperBound ValidityNoUpperBoundInAlonzoEra)
-    , txMetadata = TxMetadataNone
-    , txAuxScripts = TxAuxScriptsNone
-    , txExtraScriptData = BuildTxWith TxExtraScriptDataNone
-    , txExtraKeyWits = TxExtraKeyWitnessesNone
-    , txProtocolParams = BuildTxWith Nothing
-    , txWithdrawals = TxWithdrawalsNone
-    , txCertificates = TxCertificatesNone
-    , txUpdateProposal = TxUpdateProposalNone
-    , txMintValue = TxMintNone
-    , txScriptValidity = BuildTxWith TxScriptValidityNone
+initTx :: HeadParameters -> TxIn StandardCrypto -> ValidatedTx (AlonzoEra StandardCrypto)
+initTx HeadParameters{contestationPeriod, parties} txIn =
+  mkUnsignedTx body dats
+ where
+  body =
+    TxBody
+      (Set.singleton txIn) -- inputs
+      mempty -- collateral
+      (StrictSeq.singleton headOut) -- outputs
+      mempty -- txcerts
+      (Wdrl mempty) -- txwdrls
+      (Coin 0) -- txfee
+      (ValidityInterval SNothing SNothing) -- txvldt
+      SNothing -- txUpdates
+      mempty -- reqSignerHashes
+      mempty -- mint
+      SNothing -- scriptIntegrityHash
+      SNothing -- adHash
+      SNothing -- txnetworkid
+      --
+  dats = TxDats $ Map.singleton headDatumHash headDatum
+
+  headOut = TxOut headAddress headValue (SJust headDatumHash)
+
+  -- TODO(SN): The main Hydra Head script address. Will be parameterized by the
+  -- thread token eventually. For now, this is just some arbitrary address, as
+  -- it is also later quite arbitrary/different per Head.
+  headAddress :: Addr StandardCrypto
+  headAddress =
+    Addr
+      network
+      (ScriptHashObj $ hashScript @(AlonzoEra StandardCrypto) headScript)
+      -- REVIEW(SN): stake head funds?
```

Comment:
question: Why would we need to `stake` funds here?

Reply:
The network is secured by having a high number of delegators. Heads would likely hold quite significant amount of money, so it's quite crucial that they can delegate while the funds are locked inside a head. 

I think we are going to need a **fourth** Plutus script for Hydra :sweat_smile: ... one which can delegate all funds inside a head, but allow withdrawals of rewards proportional to each head participant because participants may not necessarily agree to which pool they want to delegate, especially if they all are SPOs :s ...

Reply:
OK, I don't know how delegating works in Cardano, need to look it up.

Reply:
We also discussed that this could be an incentive for somebody providing infrastructure for (managed) Hydra Heads. They could be allowed to decide the stake pool (likely themselves) assets are stacked. This would mean less fees, but no staking to users of the Head .. it's another mode than splitting like you describe @KtorZ :)